tensorflowonspark/TFSparkNode.py

```diff
@@ -295,20 +298,26 @@ def _mapfn(iter):
       logger.info("export TF_CONFIG: {}".format(tf_config))
       os.environ['TF_CONFIG'] = tf_config
 
-    # reserve GPU(s) again, just before launching TF process (in case situation has changed)
-    if compat.is_gpu_available():
-      # compute my index relative to other nodes on the same host (for GPU allocation)
-      my_addr = cluster_spec[job_name][task_index]
-      my_host = my_addr.split(':')[0]
-      flattened = [v for sublist in cluster_spec.values() for v in sublist]
-      local_peers = [p for p in flattened if p.startswith(my_host)]
-      my_index = local_peers.index(my_addr)
-
-      num_gpus = tf_args.num_gpus if 'num_gpus' in tf_args else 1
-      gpus_to_use = gpu_info.get_gpus(num_gpus, my_index)
-      gpu_str = "GPUs" if num_gpus > 1 else "GPU"
-      logger.debug("Requested {} {}, setting CUDA_VISIBLE_DEVICES={}".format(num_gpus, gpu_str, gpus_to_use))
-      os.environ['CUDA_VISIBLE_DEVICES'] = gpus_to_use
+    if version.parse(pyspark.__version__) >= version.parse("3.0.0"):
+      from pyspark import TaskContext
+      context = TaskContext()
+      gpus = context.resources()['gpu'] if 'gpu' in context.resources() else []
```

Comment:
You need to explicilty get the addresses out of this. context.resources()['gpu'] returns a ResourceInformation object.
so you need:
      gpus = context.resources()['gpu'].addresses if 'gpu' in context.resources() else []

Reply:
k, thx